Documentation/specs/installation-scenarios.md

```diff
@@ -0,0 +1,243 @@
+Version files can be found in multiple places:
+- Package: relative path inside the package ./.version
+- Latest/LKG version file: WIP
+
+URL:
+```
+https://dotnetcli.blob.core.windows.net/dotnet/<Channel>/<VersionPointer>.<OSID>.version
+```
+
+### File content
+Each version file contains two lines describing the build:
+```
+<CommitHash>
+<Version>
+```
+
+## Version badge
+Version badge (SVG) is an image with textual representation of `<Version>`. It can be found under following URL:
+```
+https://dotnetcli.blob.core.windows.net/dotnet/<Channel>/<VersionPointer>.<OSID>.svg
+```
+
+## Questions/gaps
+- Version Pointer links should be permanent and hosted on a separate domain
+
+# Package content
+Currently package is required to contain two files:
+- .version - [version file](#version-file)
+- dotnet<ExecutableExtension> - entry point for all dotnet commands
+
+## Disk Layout
+```
+.\
+    .version
+    bin\
+        dotnet<ExecutableExtension>
+```
+
+# Channels
+Currently we have 3 channels which gives us idea about stability and quality of our product.
+
+| Channel name | Description |
+| :---: | :--- |
+| future | This channel can contain new features which may not be fully complete. This is usually most unstable channel. |
+| preview | This channel is in the process of stablization. Most of the bugs and gaps are known. No new features expected. |
+| production | This is the most stable channel. Features and gaps are known. No breaking changes can be expected. This channel will only be producing new versions on hotfixes. |
+
+## Github branches relation
+
+Each branch on each successful build produces packages described in [build output](#build-output). Mapping between branches and channel name can be found in the table below:
+
+| Channel name | Github branch |
+| :---: | :--- |
+| future | master |
+| preview | rel/1.0.0 |
+| production | N/A, prod? |
+
+## Debian feed relation
+
+After each successful build package is being pushed to the debian feed. More information on debian feed can be found [here](#debian-feed).
+
+| Channel name | `<DebianPackageName>` |
+| :---: | :--- |
+| future | dotnet-future |
+| preview | dotnet-preview |
+| production | dotnet |
+
+## Nuget semantic version suffix relation
+WIP
+
+## Questions
+- What is the bar for triggering hotfix?
+
+# OSID
+
+OSID represents abbreviation for:
+```
+<OSName><LowestSupportedOSVersion>.<Architecture>
+```
+This naming scheme gives us flexibility to easily create new binaries when OS makes a breaking change without creating confusing names.
+
+In example, we currently put `api-ms-*.dll` files in our binaries. Those files are not needed on Windows 8 and higher. When using name `win7.x64` we can easily decide to get rid of `api-ms-*.dll` in the newest packages and simply call new version `win8.x64` which would mean that from Windows 8 forward those are recommended binaries (there is currently no issue with those files and this should be only treated as an example).

```

Comment:
Those files won't be needed regardless, won't they be part of the shared framework?